labman/db/process.py

```diff
@@ -640,23 +641,98 @@ class GDNAPlateCompressionProcess(Process):
-                        # and re-add 1 at the end
-                        out_well_row = (((well.row - 1) * 2) + row_pad) + 1
-                        out_well_col = (((well.column - 1) * 2) + col_pad) + 1
-                        out_well = container_module.Well.create(
-                            out_plate, self, volume, out_well_row,
-                            out_well_col)
-                        composition_module.CompressedGDNAComposition.create(
-                            self, out_well, volume, well.composition)
+    # basically a struct: see https://stackoverflow.com/a/36033 .
+    # all properties are 0-based.
+    # for input_plate_order_index, 0 = top-left, 1 = top-right,
+    # 2 = bottom-left, 3 = bottom-right
+    InterleavedPosition = namedtuple("InterleavedPosition",
+                                     "output_row_index output_col_index "
+                                     "input_plate_order_index input_row_index "
+                                     "input_col_index")
+
+    @staticmethod
+    def get_interleaved_quarters_position_generator(
+            num_quarters, total_num_rows, total_num_cols):
+        """Make generator of positions interleaving small plates onto large.
+
+        When smaller plates are compressed onto a larger plate (such as 1 to 4
+        96-well plates compressed onto a 384-well plate, or 1 to 4
+        384-well plates compressed onto a 1536-well plate), one strategy is to
+        interleave the input plates' wells, as shown below (for input plates
+        W, X, Y, and Z):
+
+        W X W X W X W X ...
+        Y Z Y Z Y Z Y Z ...
+        W X W X W X W X ...
+        Y Z Y Z Y Z Y Z ...
+
+        This function creates a generator that yields interleaved positions for
+        the specified number of input plates on the output plate, in the order
+        given by reading down each column of each input plate (ordered
+        from W-Z)--for example, assuming W-Z are 96-well input plates and V is
+        a 384-well output plate, this generates position mappings in the
+        following order:
+        W:A1->V:A1
+        W:B1->V:C1
+        W:C1->V:E1
+        ...
+        W:H1->V:O1
+        W:A2->V:A3
+        W:B2->V:C3
+        ...
+        W:H12->V:O23
+        X:A1->V:A2
+        X:B2->V:C2
+        ... etc.
+
+        Parameters
+        ----------
+        num_quarters : int
+            Number of quarters of interleaved positions to generate; equivalent
+            to number of smaller plates to interleave.  Must be 1-4, inclusive.
+        total_num_rows : int
+            Number of rows on large plate (e.g., 16 for a 384-well plate); must
+            be even.
+        total_num_cols : int
+            Number of columns on large plate (e.g., 24 for a 384-well plate);
+            must be even.
+
+        Yields
+        -------
+        InterleavedPosition namedtuple
+
+        Raises
+        ------
+        ValueError
```

Comment:
My _guess_ is that there is some kind of relationship between of num_quarters and total_num_rows/cols. In other words, I don't think is possible to have num_quarters = 1 and total_num_rows = total_num_cols = 0; which I _think_ are valid values for these tests, right? Thus, I guess we need to add other kind of tests to validate that relationship.

Reply:
Correct!  num_quarters=1, total_num_rows=2, total_num_cols=4 would return two positions (1/4 of 2x4) and num_quarters=1, total_num_rows=2, total_num_cols=2 would return just one position (1/4 of 2x2).